test/test-cases/typescript-css-modules/__snapshots__/typescript-css-modules.test.js.snap

```diff
@@ -27,30 +62,56 @@ export const nested: string;
+}
+`;
+
+exports[`typescript-css-modules build-ssr should create valid app 1`] = `
+Object {
+  "content": "<!DOCTYPE html><html><head>
+          <meta charset=\\"UTF-8\\">
+          <title>My Awesome Project</title>
+          <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
+          <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://localhost:4003/main-b8efa93ad3ab8918f8db.css\\">
+        </head>
+        <body>
+          <div id=\\"app\\"><div class=\\"_1brbpTo Jlnxw97\\" data-reactroot=\\"\\"><div class=\\"_3YtmnHj lwsthgY\\">Hello World</div></div></div>
+          <script type=\\"text/javascript\\" src=\\"http://localhost:4003/runtime-f6013ffd906dccb8790c.js\\" defer=\\"\\"></script>
+<script type=\\"text/javascript\\" src=\\"http://localhost:4003/vendors~main-1d95f7f495034eec0220.js\\" defer=\\"\\"></script>
+<script type=\\"text/javascript\\" src=\\"http://localhost:4003/main-012abf00716afb8575d2.js\\" defer=\\"\\"></script>
+        
+      
+    </body></html>",
+  "errors": Array [],
+  "sourceHtml": "
+      <!DOCTYPE html>
+      <html>
+        <head>
+          <meta charset=\\"UTF-8\\">
+          <title>My Awesome Project</title>
+          <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
+          <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"http://localhost:4003/main-b8efa93ad3ab8918f8db.css\\" />
+        </head>
+        <body>
+          <div id=\\"app\\"><div class=\\"_1brbpTo Jlnxw97\\" data-reactroot=\\"\\"><div class=\\"_3YtmnHj lwsthgY\\">Hello World</div></div></div>
+          <script type=\\"text/javascript\\" src=\\"http://localhost:4003/runtime-f6013ffd906dccb8790c.js\\" defer></script>
+<script type=\\"text/javascript\\" src=\\"http://localhost:4003/vendors~main-1d95f7f495034eec0220.js\\" defer></script>
+<script type=\\"text/javascript\\" src=\\"http://localhost:4003/main-012abf00716afb8575d2.js\\" defer></script>
+        </body>
+      </html>
+    ",
+  "warnings": Array [],
 }
 `;
 
 exports[`typescript-css-modules build-ssr should generate the expected files 1`] = `
 Object {
-  "index.html": "
-  <!DOCTYPE html>
-  <html>
-    <head>
-      <meta charset=\\"UTF-8\\">
-      <title>My Awesome Project</title>
-      <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
-      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/style.css\\" />
-    </head>
-    <body>
-      <div id=\\"app\\"><div class=\\"_1brbpTo Jlnxw97\\" data-reactroot=\\"\\"><div class=\\"_3YtmnHj lwsthgY\\">Hello World</div></div></div>
-      <script type=\\"text/javascript\\" src=\\"/main.js\\"></script>
-    </body>
-  </html>
-",
+  "assets.json": "{\\"main\\":{\\"js\\":[\\"http://localhost:4003/runtime-f6013ffd906dccb8790c.js\\",\\"http://localhost:4003/vendors~main-1d95f7f495034eec0220.js\\",\\"http://localhost:4003/main-012abf00716afb8575d2.js\\"],\\"css\\":\\"http://localhost:4003/main-b8efa93ad3ab8918f8db.css\\"}}",
```

Comment:
@mattsjones why is the vendors chuckfile `vendors~main-[hash]` and not just `vendors-[hash]` (does vendors split out all node_modules imports from src js)?

Also, an idea around the main filename/entry point could be that the main js file could be `scripts-[hash].js` with the main css file being `styles-[hash]`.

I think this was a pattern in html boilerplate but maybe the community has moved on since then? Just an idea anyway :-)

Reply:
The names are generated by webpack. Essentially, webpack looks at your app and creates many chunks. Then it combines them based off what is shared between them. The names are what entries (+ vendor) share this specific asset. If your app is big enough you may end up with multiple `vendor` chunks. e.g. `vendors~home~details-[hash]` and `vendors~home-[hash]`.

On prepending `scripts/styles` I don't think there's much value. All these assets are static and should be deployed in the same way. Their type is inferred by their extension.

Reply:
no probs. thanks for clarifying. On the `scripts/styles`, I wasn't suggesting we prepend but rather rename the main assets accordingly however this explanation of the webpack chunking voids the idea :-)